src/Config/RuntimeConfigLoader.cs

```diff
@@ -145,7 +145,9 @@ public RuntimeConfigLoader(string? connectionString = null)
             }
 
         }
-        catch (JsonException ex)
+        catch (Exception ex) when (
+            ex is JsonException ||
+            ex is DataApiBuilderException)
```

Comment:
if the exception is DataApiBuilderException, is the error message here still a valid message? How is the deserialization failure message valid for a connection string failure?

Reply:
I thought it was still descriptive for what is happening because this is a post processing step of the deserialization. We deserialize directly from the json string to create the objects, then we have a conditional update as part of the post processing. So in my eyes that was still part of the deserialization process and giving it another name in the error messaging might be confusing.

I considered calling this "parsing" instead but thought it was clearer like this. this is what it looks like now

![image](https://github.com/Azure/data-api-builder/assets/93220300/35395f5e-7517-4d17-9a95-593a110441a3)

Reply:
include that screenshot in the pr description to demonstrate this PR's affect.